solr/core/src/java/org/apache/solr/response/transform/DeeplyNestedChildDocTransformer.java

```diff
@@ -108,12 +103,11 @@ public void transform(SolrDocument rootDoc, int rootDocId) {
       final LeafReaderContext leafReaderContext = searcher.getIndexReader().leaves().get(seg);
       final SortedDocValues segPathDocValues = DocValues.getSorted(leafReaderContext.reader(), NEST_PATH_FIELD_NAME);
 
-      Multimap<String,SolrDocument> pendingParentPathsToChildren = ArrayListMultimap.create();
+      Map<String, Multimap<String, SolrDocument>> pendingParentPathsToChildren = new HashMap<>();
```

Comment:
Ah good catch; yes a Map and not MultiMap at outer level once we add the label intermediate